src/shortestpaths/desopo-pape.jl

```diff
@@ -0,0 +1,46 @@
+"""
+    struct DEposoPapeState{T, U}
+
+An [`AbstractPathState`](@ref) designed for D`Esopo-Pape shortest-path calculations.
+"""
+struct DEsopoPapeState{T <:Real, U <: Integer} <: AbstractPathState
+    parents::Vector{U}
+    dists::Vector{T}
+end
+
+function DEsopoPape_shortest_path(g::AbstractGraph, 
+    src::U,
+    distmx::AbstractMatrix{T} = weights(g)) where T <: Real where U <: Integer
+    
+    nvg = nv(g)
+    dists = fill(typemax(T), nvg)
+    parents = zeros(U, nvg)
+    state = fill(2, nvg)
```

Comment:
consider using a `Vector{UInt8}` or `Vector{Int8}` instead of the Vector{Int} here.

Reply:
for `state` , right?

Reply:
Yes